### PR TITLE
Update global atmos vminmon job to use COMIN/COMOUT

### PR DESCRIPTION
**Pull request by mingshichen-noaa**
# Description
 
NCO has requested that each COM variable specify whether it is an input or an output. This completes that process for the global minimization monitor job.

Refs https://github.com/NOAA-EMC/global-workflow/issues/2451

# Type of change

- Maintenance (code refactor, clean-up, new CI test, etc.)

# Change characteristics
- Is this a breaking change (a change in existing functionality)? NO
- Does this change require a documentation update? NO

# How has this been tested?
 
- Clone and build on RDHPCS
- Cycled tests on Hercules
- Forecast-only tests on Hercules
 
# Checklist
- [x] Any dependent changes have been merged and published
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] New and existing tests pass with my changes
- [x] I have made corresponding changes to the documentation if necessary

### CODE REVIEW

**Review by aerorahul — DISMISSED** [review]
looks ok

File: jobs/JGLOBAL_ATMOS_VMINMON [hunk 1]
```diff
@@ -17,16 +17,16 @@ export gcyc=${GDATE:8:2}
 #############################################
 # TANKverf - WHERE OUTPUT DATA WILL RESIDE
 #############################################
-YMD=${PDY} HH=${cyc} declare_from_tmpl -rx COM_ATMOS_ANALYSIS
-YMD=${PDY} HH=${cyc} declare_from_tmpl -rx COM_ATMOS_MINMON
-YMD=${gPDY} HH=${gcyc} declare_from_tmpl -rx COM_ATMOS_MINMON_PREV:COM_ATMOS_MINMON_TMPL
+YMD=${PDY} HH=${cyc} declare_from_tmpl -rx  COMIN_ATMOS_ANALYSIS:COM_ATMOS_ANALYSIS_TMPL
+YMD=${PDY} HH=${cyc} declare_from_tmpl -rx  COMOUT_ATMOS_MINMON:COM_ATMOS_MINMON_TMPL
```
Comment (KateFriedman-NOAA) [inline jobs/JGLOBAL_ATMOS_VMINMON]: An extra space was added on both lines 20 and 21 after `-rx`. Please remove those, thanks!

**Review by KateFriedman-NOAA — CHANGES_REQUESTED** [review]

File: jobs/JGLOBAL_ATMOS_VMINMON (hunk 1 above)
Comment (mingshichen-noaa) [reply jobs/JGLOBAL_ATMOS_VMINMON]: @KateFriedman-NOAA 
Yes, I removed the extra space.
Thanks.

File: jobs/JGLOBAL_ATMOS_VMINMON [hunk 2]
```diff
@@ -17,16 +17,16 @@ export gcyc=${GDATE:8:2}
 #############################################
 # TANKverf - WHERE OUTPUT DATA WILL RESIDE
 #############################################
-YMD=${PDY} HH=${cyc} declare_from_tmpl -rx COM_ATMOS_ANALYSIS
-YMD=${PDY} HH=${cyc} declare_from_tmpl -rx COM_ATMOS_MINMON
-YMD=${gPDY} HH=${gcyc} declare_from_tmpl -rx COM_ATMOS_MINMON_PREV:COM_ATMOS_MINMON_TMPL
+YMD=${PDY} HH=${cyc} declare_from_tmpl -rx COMIN_ATMOS_ANALYSIS:COM_ATMOS_ANALYSIS_TMPL
+YMD=${PDY} HH=${cyc} declare_from_tmpl -rx COMOUT_ATMOS_MINMON:COM_ATMOS_MINMON_TMPL
```
Comment (WalterKolczynski-NOAA) [inline jobs/JGLOBAL_ATMOS_VMINMON]: Does the same thing, but a bit easier to read and conforms with other jobs.
```suggestion
YMD=${PDY} HH=${cyc} declare_from_tmpl -rx \
    COMIN_ATMOS_ANALYSIS:COM_ATMOS_ANALYSIS_TMPL \
    COMOUT_ATMOS_MINMON:COM_ATMOS_MINMON_TMPL
```

File: jobs/JGLOBAL_ATMOS_VMINMON [hunk 3]
```diff
@@ -17,16 +17,16 @@ export gcyc=${GDATE:8:2}
...
+YMD=${gPDY} HH=${gcyc} declare_from_tmpl -rx COMOUT_ATMOS_MINMON_PREV:COM_ATMOS_MINMON_TMPL
```
Comment (WalterKolczynski-NOAA) [inline jobs/JGLOBAL_ATMOS_VMINMON]: I checked the scripts. This is only used as an input.

File: jobs/JGLOBAL_ATMOS_VMINMON [hunk 4]
```diff
@@ -17,16 +17,16 @@ export gcyc=${GDATE:8:2}
...
-export M_TANKverf=${M_TANKverf:-${COM_ATMOS_MINMON}}
-export M_TANKverfM1=${M_TANKverfM1:-${COM_ATMOS_MINMON_PREV}}
+export gsistat="${COMIN_ATMOS_ANALYSIS}/${RUN}.t${cyc}z.gsistat"
+export M_TANKverf=${M_TANKverf:-${COMOUT_ATMOS_MINMON}}
+export M_TANKverfM1=${M_TANKverfM1:-${COMOUT_ATMOS_MINMON_PREV}}
 
-if [[ ! -d ${M_TANKverf} ]]; then mkdir -p -m 775 "${M_TANKverf}" ; fi
-if [[ ! -d ${M_TANKverfM1} ]]; then mkdir -p -m 775 "${M_TANKverfM1}" ; fi
+if [[ ! -d ${M_TANKverf} ]]; then mkdir -p "${M_TANKverf}" ; fi
+if [[ ! -d ${M_TANKverfM1} ]]; then mkdir -p "${M_TANKverfM1}" ; fi
```
Comment (WalterKolczynski-NOAA) [inline jobs/JGLOBAL_ATMOS_VMINMON:32]: Since this is only an input, we shouldn't be creating it.
```suggestion
```

**Review by WalterKolczynski-NOAA — CHANGES_REQUESTED** [review]

File: jobs/JGLOBAL_ATMOS_VMINMON (hunk 3 above)
Comment (WalterKolczynski-NOAA) [reply jobs/JGLOBAL_ATMOS_VMINMON]:
```suggestion
YMD=${gPDY} HH=${gcyc} declare_from_tmpl -rx COMIN_ATMOS_MINMON_PREV:COM_ATMOS_MINMON_TMPL
```

File: jobs/JGLOBAL_ATMOS_VMINMON (hunk 4 above)
Comment (mingshichen-noaa) [reply jobs/JGLOBAL_ATMOS_VMINMON:32]: @WalterKolczynski-NOAA
From scripts ( exglobal_atmos_vminmon.sh), these three directory variables should be input directories. Hence line 28 and 29 should be removed. Am I right?

File: jobs/JGLOBAL_ATMOS_VMINMON (hunk 2 above)
Comment (mingshichen-noaa) [reply jobs/JGLOBAL_ATMOS_VMINMON]: @WalterKolczynski-NOAA 
Yes, I will change them.

File: jobs/JGLOBAL_ATMOS_VMINMON (hunk 4 above)
Comment (WalterKolczynski-NOAA) [reply jobs/JGLOBAL_ATMOS_VMINMON:32]: Only `$M_TANKverfM1` is an input. `$M_TANKverif` is an output (the perl scripts write to it).

Comment (mingshichen-noaa) [reply jobs/JGLOBAL_ATMOS_VMINMON:32]: @WalterKolczynski-NOAA 
After reading 3 perl files in ush (minmon_xtrct_costs.pl  minmon_xtrct_gnorms.pl  minmon_xtrct_reduct.pl),    gsistat should be only an input file defined in jobs/JGLOBAL_ATMOS_VMINMON. This implies the following modifications:
 20 YMD=${PDY} HH=${cyc} declare_from_tmpl -rx \
 21         COMIN_ATMOS_ANALYSIS:COM_ATMOS_ANALYSIS_TMPL \
 22         COMOUT_ATMOS_MINMON:COM_ATMOS_MINMON_TMPL
 23
 24 YMD=${gPDY} HH=${gcyc} declare_from_tmpl -rx \
 25         COMOUT_ATMOS_MINMON_PREV:COM_ATMOS_MINMON_TMPL
 26
 27 export gsistat="${COMIN_ATMOS_ANALYSIS}/${RUN}.t${cyc}z.gsistat"
 28 export M_TANKverf=${M_TANKverf:-${COMOUT_ATMOS_MINMON}}
 29 export M_TANKverfM1=${M_TANKverfM1:-${COMOUT_ATMOS_MINMON_PREV}}
 30
 31 if [[ ! -d ${M_TANKverf} ]]; then mkdir -p "${M_TANKverf}" ; fi
 32 if [[ ! -d ${M_TANKverfM1} ]]; then mkdir -p "${M_TANKverfM1}" ; fi
 33

Comment (WalterKolczynski-NOAA) [reply jobs/JGLOBAL_ATMOS_VMINMON:32]: What modifications are you proposing? That looks like what is there now.

**Pull request closed without merging**